pkg/common/config/client_option.go

```diff
@@ -25,6 +25,13 @@ import (
 	"github.com/cloudwego/hertz/pkg/protocol/consts"
 )
 
+type HostClientState interface {
```

Comment:
🤔 I don't want to expose connection pool to users and I think there is no other requirements temporarily. Besides, I don't think add method will cause break change because most users just use the interface we provide.
If you have concrete better implementation, you can tell me.

Reply:
I think we can refer to https://pkg.go.dev/database/sql#DB.Stats

Reply:
What does efectn think? @efectn

Reply:
I think we can just expose some useful pool status info like sql instead of exposing pool. I think it's not good idea to expose pool, too.